### DISCUSSION TITLE
Provide cleaner examples

### DISCUSSION THREAD

**Discussion by MrAmericanMike** [trpc/trpc#2325] [General]
The general idea of this post is to explain why I think the examples should be stripped down to a bare minimum, or at least how I fell about the examples provided for an Express/tRPC app.

*****
So from a newbie point of view, tRPC looks very intimidating, but I came to the conclusion this is because of the amount of complexity and boilerplate that the examples are trying to cover.

I learn with time that the simpler the introduction to a new technology is, the better the experience (There is always time to add on top in the future)

Going to the most simple Express/tRPC example and looking at the client.ts I see a lot of boilerplate and intimidating code.

In the provided CodeSandbox example (Not to mention that it crashes out of the box after trying on Brave and Chrome) 

https://codesandbox.io/s/github/trpc/trpc/tree/main/examples/express-server

I only got it to work after forking (No idea why)

Why to have so much stuff on the client.ts when for an introduction to someone new maybe just this would be better:

```js
import { createTRPCClient } from "@trpc/client";
import AbortController from "abort-controller";
import fetch from "node-fetch";
import type { AppRouter } from "./server";

global.AbortController = AbortController;
global.fetch = fetch as any;

async function main() {
  const client = createTRPCClient<AppRouter>({
    url: "http://localhost:2021/trpc"
  });

  const hello = await client.query("hello");
  console.log(hello);
}

main();
```

I understand the polyfill (at least I believe I do) but maybe a better comment saying "Adding polyfill to make the code find fetch as it would in a client side app" Or whatever it's really happening. Or even better, create an actual HTML index page and serve that from the server side using express and write the client code in some script.ts (I understand this would require to add all the necessary pieces to convert that back into js, which isn't a bad thing IMO)

*****

On the server side, again, I understand the example tries to cover a lot, but to start with, I personally would remove "zod" from it, as I said, show me the basic functionality, don't confuse me with extra layers, you can do that in another example with some of this "more advanced" features.

The part that confuses me the most is...

```js
const createContext = ({
  req,
  res,
}: trpcExpress.CreateExpressContextOptions) => {
  const getUser = () => {
    if (req.headers.authorization !== 'secret') {
      return null;
    }
    return {
      name: 'alex',
    };
  };

  return {
    req,
    res,
    user: getUser(),
  };
};
```

This is just UNREADABLE. Like what in the World is going on there?

if `createContext: () => null` works, for a simple example give me that, we can dig into what the contexts are later on and as we need them.

I understand the library is trying to show us all the potential it has (well not all, but more than it's needed to just get going with it)

Would love to see examples with the bare minimum, a Express Server, Serve a public folder, maybe have it rendering a button that you click and it triggers a fetch.

**Comment by juliusmarminge** [comment]
For the next-branch, there is a minimal example for using with Next.js (see #2169 or repo [here](https://github.com/juliusmarminge/next-minimal-trpc-v10)).

Would something like this, but for usage with express+vanillajs be what you are looking for?
